pkg/middleware/jwt_session.go

```diff
@@ -0,0 +1,168 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/coreos/go-oidc"
+	"github.com/justinas/alice"
+	middlewareapi "github.com/oauth2-proxy/oauth2-proxy/pkg/apis/middleware"
+	sessionsapi "github.com/oauth2-proxy/oauth2-proxy/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
+)
+
+const jwtRegexFormat = `^eyJ[a-zA-Z0-9_-]*\.eyJ[a-zA-Z0-9_-]*\.[a-zA-Z0-9_-]+$`
```

Comment:
What does the `eyJ` really buy us? I think the `[a-zA-Z0-9_-]` + `\.` is the key part to confirm we get a URL base64 with the 3 JWT parts split by `.`.

I know `eyJ` is supposed to represent `{"` -- but `eyJ` represents 18 bytes, the first 2 bytes of the 3rd character to encode affect the `J` part (I guess its fine that [a-zA-Z] all result in J...? but 0-9 doesn't for instance.)

I think we ditch it in case some weird provider implementations that break a spec roll on in.

Reply:
Hmm, I guess this was mostly used for catching JWTs that came in via `Authorization: Basic ...`? If we do change this -- maybe just to `ey` -- that for sure aligns with `{"`

Reply:
These are all good questions, I simply copied this from where it was in the main package, I don't really recall before why we had it this way. 

Happy to look into changing it, but maybe we do that in a separate PR so we can keep the scope of this one smaller?

>I guess its fine that [a-zA-Z] all result in J...? but 0-9 doesn't for instance.
I wonder if claim names are allowed to start with numbers 🤔

Reply:
> I wonder if claim names are allowed to start with numbers 🤔

Cause we've never seen an OAuth Provider be a special snowflake and break the spec 🤣